File: client/manager/src/model/feedback/attachment.ts
export class AddAttachmentReq {
  constructor(public data: Uint8Array, public fileName: string, public fileType: string, public fileSize: number) {
    this.data = data
    this.fileName = fileName
    this.fileType = fileType
    this.fileSize = fileSize
  }
}

export class Attachment {
  constructor(
    public id: number,
    public feedbackId: number,
    public fileName: string,
    public fileType: string,
    public fileSize: number,
    public fileUrl: string
  ) {
    this.id = id
    this.feedbackId = feedbackId
    this.fileName = fileName
    this.fileType = fileType
    this.fileSize = fileSize
    this.fileUrl = fileUrl
  }
}

File: server/internal/feedback/services/dto/attachment.go
package dto

import (
	"server/common"
	"server/common/base/dto"
	"server/common/converter"
)

type AttachmentDTO struct {
	dto.BaseDTO
	FeedbackID uint64 `json:"feebackId"`
	FileUrl    string `json:"fileUrl"`
	FileType   string `json:"fileType"`
	FileName   string `json:"fileName"`
	FileSize   int    `json:"fileSize"`
}

type AddAttachmentDTO struct {
	FeedbackID uint64  `json:"feedbackId"`
	Data       []uint8 `json:"data"`
	FileType   string  `json:"fileType"`
	FileName   string  `json:"fileName"`
	FileSize   int     `json:"fileSize"`
}

func (d *AddAttachmentDTO) ToAttachmentDTO() *AttachmentDTO {
	feedbackDTO := &AttachmentDTO{}
	converter.Copy(feedbackDTO, d)
	feedbackDTO.CreatedBy = common.GetLoginUserID()
	feedbackDTO.UpdatedBy = common.GetLoginUserID()
	return feedbackDTO
}
